src/types.ts

```diff
@@ -1,122 +1,86 @@
-import { Surreal } from "./surreal.ts";
-import { UUID } from "./library/cbor/uuid.ts";
+import { type RecordId, Uuid } from "./data";
 
-export const UseOptions = z.object({
-	namespace: z.coerce.string(),
-	database: z.coerce.string(),
-});
+export type ActionResult<T extends Record<string, unknown>> = Prettify<
+	T["id"] extends RecordId ? T : { id: RecordId } & T
+>;
 
-export type UseOptions = z.infer<typeof UseOptions>;
-
-export type ActionResult<
-	T extends Record<string, unknown>,
-> = Prettify<T["id"] extends RecordId ? T : { id: RecordId } & T>;
-
-export type Prettify<T> =
-	& {
-		[K in keyof T]: T[K];
-	}
-	// deno-lint-ignore ban-types
-	& {};
+export type Prettify<T> = {
+	[K in keyof T]: T[K];
+} & {}; // deno-lint-ignore ban-types
 
 //////////////////////////////////////////////
 //////////   AUTHENTICATION TYPES   //////////
 //////////////////////////////////////////////
 
-export const SuperUserAuth = z.object({
-	namespace: z.never().optional(),
-	database: z.never().optional(),
-	scope: z.never().optional(),
-	username: z.coerce.string(),
-	password: z.coerce.string(),
-});
-
-export type SuperUserAuth = z.infer<typeof SuperUserAuth>;
-
-export const NamespaceAuth = z.object({
-	namespace: z.coerce.string(),
-	database: z.never().optional(),
-	scope: z.never().optional(),
-	username: z.coerce.string(),
-	password: z.coerce.string(),
-});
-
-export type NamespaceAuth = z.infer<typeof NamespaceAuth>;
-
-export const DatabaseAuth = z.object({
-	namespace: z.coerce.string(),
-	database: z.coerce.string(),
-	scope: z.never().optional(),
-	username: z.coerce.string(),
-	password: z.coerce.string(),
-});
-
-export type DatabaseAuth = z.infer<typeof DatabaseAuth>;
-
-export const ScopeAuth = z.object({
-	namespace: z.coerce.string().optional(),
-	database: z.coerce.string().optional(),
-	scope: z.coerce.string(),
-}).catchall(z.unknown());
-
-export type ScopeAuth = z.infer<typeof ScopeAuth>;
-
-export const AnyAuth = z.union([
-	SuperUserAuth,
-	NamespaceAuth,
-	DatabaseAuth,
-	ScopeAuth,
-]);
-export type AnyAuth = z.infer<typeof AnyAuth>;
-
-export const Token = z.string({ invalid_type_error: "Not a valid token" });
-export type Token = z.infer<typeof Token>;
-
-export const TransformAuth = z.union([
-	z.object({
-		namespace: z.string().optional(),
-		database: z.string().optional(),
-		scope: z.never().optional(),
-		username: z.string(),
-		password: z.string(),
-	}).transform(({
-		namespace,
-		database,
-		username,
-		password,
-	}) => {
-		const vars: Record<string, unknown> = {
-			user: username,
-			pass: password,
-		};
-
-		if (namespace) {
-			vars.ns = namespace;
-			if (database) {
-				vars.db = database;
-			}
+export function convertAuth(params: AnyAuth): Record<string, unknown> {
+	const cloned: Record<string, unknown> = { ...params };
+	const convertString = (a: string, b: string, optional?: boolean) => {
+		if (a in params) {
+			cloned[b] = `${cloned[a]}`;
+			delete cloned[a];
```

Comment:
This can be bad for performance due to changing the object shape

Reply:
Would initiating a new object be more performant?

Reply:
Not necessarily, it depends on the pattern of the data

Reply:
It basically rewrites: 
- `namespace` -> `ns`
- `database` -> `db`
- `scope` -> `sc`
- `access` -> `ac`